### PR TITLE
Lando xhprof settings.

### PR DESCRIPTION
**Pull request by sherakama**
# Adds XHProf to Lando environment.

Super low priority. In fact, you can ignore it forever if you'd like.

### CODE REVIEW

**Comment by sherakama** [comment]
Not sure if this is helpful or not to add to this project but here it is anyway. Close this PR if it doesn't belong.

**Comment by sherakama** [comment]
Added @jbickar & @mdyoung3 

Have a look and see if there is anything useful in here for you. If not close and delete this PR.

**Comment by mdyoung3** [comment]
@sherakama Do you need the module to run it? https://www.drupal.org/project/xhprof

Could add that to the composer file as `require-dev`.

**Comment by sherakama** [comment]
Yeah, there is some configuration on the Drupal side of things to get this to work but I intentionally stopped short of that in this PR. I only wanted to provide the ability to run xhprof inside the LANDO container and then any Drupal side configuration would be up to the developer.

**Comment by jbickar** [comment]
@mdyoung3 , could we add `xhprof` to the Docksal config? We do not currently have Lando config on this repo.

**Comment by jbickar** [comment]
> you can ignore it forever if you'd like.

@sherakama, careful: you know that I can outwait you.

File: soe/Gruntconfig.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   },
   "eslint": true,
   "drush": {
-    "path": "/Users/boznik/.composer/vendor/bin/drush",
+    "path": "./vendor/bin/drush",
```
Comment (sherakama) [inline soe/Gruntconfig.json:23]: If you take only one thing from this PR let it be this line.

**Comment by sherakama** [comment]
> > you can ignore it forever if you'd like.
> 
> @sherakama, careful: you know that I can outwait you.

Don't forget I train as an endurance athlete.

**Comment by mdyoung3** [comment]
@jbickar Docksal comes with xhprof disabled by default, but you can enable it in an ENV file. My guess would be we wouldn't want to enable it by default. Given how much of a performance hit you'll take. 

Probably what I'd recommend is using the command that enables it when you need to do some debugging. And then disable it after. If xhprof is used a lot, we might want to consider adding ~~it~~ the module to either be downloaded via composer or many see if we can download it with Grunt. Then maybe added to .gitignore so it doesn't end up on Acquia. 

`fin config set --env=local XHPROF_ENABLED=1`

(That flag setting env to local creates the local ENV file too! I did not know that.)

Source: https://docs.docksal.io/tools/xhprof/

File: soe/Gruntconfig.json (hunk 1 above)
Comment (mdyoung3) [reply soe/Gruntconfig.json:23]: Ah man. I just renamed my mac user to `boznik`.

**Comment by jbickar** [comment]
> @jbickar Docksal comes with xhprof disabled by default, but you can enable it in an ENV file. My guess would be we wouldn't want to enable it by default. Given how much of a performance hit you'll take.
> 
> Probably what I'd recommend is using the command that enables it when you need to do some debugging. And then disable it after. If xhprof is used a lot, we might want to consider adding it to either be downloaded via composer or many see if we can download it with Grunt. Then maybe added to .gitignore so it doesn't end up on Acquia.
> 
> `fin config set --env=local XHPROF_ENABLED=1`
> 
> (That flag setting env to local creates the local ENV file too! I did not know that.)
> 
> Source: https://docs.docksal.io/tools/xhprof/

@mdyoung3 , that sounds like another PR with the  path to drush changes from this PR, and documentation in the Docksal README about XHPROF. We can revisit for Lando. I am learning things about running both, but haven't put together my notes yet.

Do as I say, not as I do. 😆

**Comment by jbickar** [comment]
> > > you can ignore it forever if you'd like.
> > 
> > 
> > @sherakama, careful: you know that I can outwait you.
> 
> Don't forget I train as an endurance athlete.

I can sit on a PR `:alot:` longer than you can run.

File: soe/Gruntconfig.json (hunk 1 above)
Comment (boznik) [reply soe/Gruntconfig.json:23]: don't forget to update the README: https://github.com/SU-SOE/grunt-drupal-tasks/blob/landoxhprof/docs/soe_build_notes.md

**Pull request closed without merging**